__tests__/app.test.js

```diff
@@ -147,6 +143,39 @@ describe("GET/api/articles/:article_id", () => {
         expect(res.body.message).toBe("Bad Request");
       });
   });
+  test("response with the status of 200 and array of all the available comments for the given article", () => {
+    return request(app)
+      .get("/api/articles/3/comments")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.comments).toBeInstanceOf(Array);
+        expect(body.comments).toBeSortedBy("created_at");
+        body.comments.forEach((comment) => {
+          expect(comment).toMatchObject({
+            comment_id: expect.any(Number),
+            title: expect.any(String),
+            article_id: expect.any(Number),
+            created_at: expect.any(String),
+            votes: expect.any(Number),
+            body: expect.any(String),
+          });
+        });
+      });
+  });
+  test("response with the status of 404 and an appropriate error message if invalid id is used", () => {
+    return request(app)
+      .get("/api/articles/9999/comments")
+      .expect(404)
+      .then((res) => {
+        expect(res.body.message).toBe("Not Found");
+      });
+  });
+  test("response with the status of 400 and an appropriate error message if parameter id is not an id ", () => {
+    return request(app)
+      .get("/api/articles/notAnId/comments")
+      .expect(400)
+      .then((res) => {
+        expect(res.body.message).toBe("Bad Request");
+      });
+  });
 });
-
```

Comment:
One very important test which is missing here is a test to ensure when given a valid article id but one with no comments associated with it we receive an empty array. The endpoint for this will be /api/articles/2/comments